src/server/accounting.c

```diff
@@ -1326,43 +1063,29 @@ account_jobend(job *pjob, char *used, int type)
 	}
 
 	if (used != NULL) {
-		i = strlen(used) + 1;
-		if (i > len)
-			if (grow_acct_buf(&pb, &len, i) == -1)
-				goto writeit;
-		(void)strcat(pb, " ");
-		(void)strcat(pb, used);
-		i = strlen(pb);
-		pb  += i;
-		len -= i;
+		if (!pbs_strcat(&acct_buf, &acct_bufsize, " "))
+			goto writeit;
+		if (!pbs_strcat(&acct_buf, &acct_bufsize, used))
```

Comment:
It will be good if you use snprintf (tempstr,...) then pbs_strcat instead of separate pbs_strcat above.

Reply:
We can use snprintf(tempstr,...) but we don't know how many resources_used entries would be there in "used" and for that we can not predict the size and increase tempstr's size so I have used pbs_strcat() so that it can increase the size accordingly.